larpix/io/multizmq_io.py

```diff
@@ -77,55 +86,33 @@ def stop_listening(self):
         for receiver in self.receivers.values():
             receiver.setsockopt(zmq.UNSUBSCRIBE, b'')
 
-    @classmethod
-    def is_valid_chip_key(cls, key):
-        '''
-        Valid chip keys must be strings formatted as:
-        ``'<daq_address>/<io_chain>/<chip_id>'``
-        Note that ``/`` is a special character and should not be used
-        in daq address
-
-        '''
-        if not super(MultiZMQ_IO, cls).is_valid_chip_key(key):
-            return False
-        if not isinstance(key, str):
-            return False
-        parsed_key = key.split('/')
-        if not len(parsed_key) == 3:
-            return False
-        try:
-            _ = int(parsed_key[1])
-            _ = int(parsed_key[2])
-        except ValueError:
-            return False
-        return True
-
-    @classmethod
-    def parse_chip_key(cls, key):
+    def parse_chip_key(self, key):
         '''
         Decodes a chip key into ``'chip_id'``, ``'io_chain'``, and ``'address'``
 
         :returns: ``dict`` with keys ``('chip_id', 'io_chain', 'addresss')``
         '''
         return_dict = {}
-        if not MultiZMQ_IO.is_valid_chip_key(key):
-            raise ValueError('invalid MultiZMQ_IO key: {}'.format(key))
-        parsed_key = key.split('/')
-        return_dict['chip_id'] = int(parsed_key[2])
-        return_dict['io_chain'] = int(parsed_key[1])
-        return_dict['address'] = parsed_key[0]
+        return_dict['chip_id'] = key.chip_id
+        return_dict['io_chain'] = key.io_channel - 1
+        if not key.io_group in self._io_group_table:
+            raise KeyError('unspecified io group {}'.format(key.io_group))
+
+        return_dict['address'] = self._io_group_table[key.io_group]
+        if not return_dict['address'] in self.receivers:
```

Comment:
You should use the ``not in`` operator like you do [here](https://github.com/larpix/larpix-control/pull/158/files#diff-c5b11bae29ad81417b440a21cf91a722R37) rather than ``not X in Y```